module/zfs/dsl_dataset.c

```diff
@@ -2273,8 +2273,6 @@ dsl_dataset_sync_done(dsl_dataset_t *ds, dmu_tx_t *tx)
 
 	ASSERT(!dmu_objset_is_dirty(os, dmu_tx_get_txg(tx)));
```

Comment:
You might consider moving this ASSERT() to the end of the function.

Reply:
Done!